Source/src/core/GameObject.h

```diff
@@ -77,6 +77,11 @@ namespace Hachiko
             return components;
         }
 
+        [[nodiscard]] ComponentTransform* const GetTransform() const 
```

Comment:
```suggestion
        [[nodiscard]] const ComponentTransform* GetTransform() const 
```

Reply:
Actually this was on purpose since this was made to avoid unnecessary lookup as previous version of engine had `game_object.GetComponent<ComponentTransform>` which returns `ComponentTransform*`. Therefore, this API should let user to tweak the values of `ComponentTransform`. Because of that, I wanted to have the `ComponentTransform*` returned by `GetTransform` to be read\write-able while prohibiting `= new ComponentTransform()` calls on it for example. That's why it was made `ComponentTransform* const` instead of `const ComponentTransform*`.

Reply:
if that is the case one of those two consts has to go

Reply:
As we talked, `ComponentTransform* const GameObject::GetTransform() const` seems to be weirdly equivalent to `ComponentTransform* GameObject::GetTransform() const` so I'm converting it to the latter.

Reply:
a typical getter that some tools like Resharper generate as in the form
`[[nodiscard]] const ComponentTransform* GetTransform() const ` that's why I suggested it, consider if that would be useful

Reply:
Yeah in our case it's not useful for us since we need stuff like `game_object->GetTransform()->SetPosition(0.0f, 0.0f, 0.0f)` for example.